modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -1473,6 +1473,21 @@ public static string Right(this string instance, int pos)
             return instance.Substring(pos, instance.Length - pos);
         }
 
+        /// <summary>
+        /// Decodes a hexadecimal string.
+        /// </summary>
+        /// <param name="instance">The hexadecimal string.</param>
+        ///  <returns>The byte array representation of this string.</returns>
+        public static byte[] HexDecode(this string instance)
+        {
+            int len = instance.Length / 2;
+            byte[] ret = new byte[len];
+            for (int i = 0; i < len; i++)
+            {
+                ret[i] = (byte)int.Parse(instance.Substring(1 * 2, 2), NumberStyles.HexNumber);
```

Comment:
This doesn't seem correct, you are always parsing the same part of the string for each byte.

Reply:
oops. i havent managed to test the c# side yet.